pkg/controller/operators/catalog/operator.go

```diff
@@ -363,6 +364,10 @@ func (o *Operator) syncSourceState(state grpc.SourceState) {
 		if err := o.catsrcQueueSet.Requeue(state.Key.Namespace, state.Key.Name); err != nil {
 			o.logger.WithError(err).Info("couldn't requeue catalogsource from catalog status change")
 		}
+		return
+	}
+	if err := o.catsrcQueueSet.Requeue(state.Key.Namespace, state.Key.Name); err != nil {
```

Comment:
isn't this just requeuing after ever state change? is that desired?

Reply:
I actually do so that the status stays in sync with what the grpc connection is reporting. This only adds additional requeues when the connection is reported as ready as requeues were already being done otherwise. I could refactor since a lot of the above is similar if you agree.

Reply:
I went ahead and refactored it.